components/go/pkg/api/api_client.go

```diff
@@ -0,0 +1,127 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	config "github.com/chef/chef-licensing/components/go/pkg/config"
+	// licenseclient "github.com/progress-platform-services/chef-platform-license-management/public/client/license-management/license"
+)
+
+const (
+	CLIENT_VERSION = "v1"
+)
+
+type APIClient struct {
+	URL        string
+	HTTPClient *http.Client
+	Headers    map[string]string
+}
+
+var apiClient *APIClient
+
+func (c *APIClient) BaseURL() string {
+	return fmt.Sprintf("%s/%s/", c.URL, CLIENT_VERSION)
+}
+
+// func NewAPIClient() licenseclient.LicenseClient {
+// 	cfg := config.GetConfig()
+// 	conf := pconfig.DefaultHttpConfig(cfg.LicenseServerURL)
+// 	logger, err := plogger.NewLogger(plogger.LoggerConfig{LogLevel: "debug", LogToStdout: true})
+// 	fmt.Println("Loggflaslfjalskfjlaksfjlkj")
+// 	logger.Warn("Test log")
+// 	if err != nil {
+// 		log.Fatal("Unable to create a logger", err)
+// 	}
+// 	agent, err := pagent.NewAgent(conf, pagent.BasicClient, pauthtype.NoAuth{}, pcache.NewLocalCache(time.Second*60, time.Second*60), logger)
+// 	if err != nil {
+// 		log.Fatal("Unable to create the api client ", err)
+// 	}
+// 	return licenseclient.NewLicenseClient(conf, agent, nil, false, nil, nil, logger)
+// }
+
+func NewClient() *APIClient {
+	cfg := config.GetConfig()
+
+	apiClient = &APIClient{
+		URL:        cfg.LicenseServerURL,
+		HTTPClient: &http.Client{},
+		Headers: map[string]string{
+			"Content-Type": "application/json",
+		},
+	}
+	return apiClient
+}
+
+func GetClient() *APIClient {
```

Comment:
you can use `sync.Once` for thread-safe initialization
example


```

var (
	apiClient *APIClient
	once      sync.Once
)

func GetClient() *APIClient {
	once.Do(func() {
		apiClient = NewClient()
	})

	return apiClient
}

```

Reply:
Done